controllers/hibp.js

```diff
@@ -20,6 +22,27 @@ async function notify (req, res) {
 }
 
 
+async function breaches (req, res, next) {
+  const clientMostRecentBreachDateTime = new Date(req.headers["if-modified-since"]);
+  const serverMostRecentBreachDateTime = req.app.locals.mostRecentBreachDateTime;
+  if (clientMostRecentBreachDateTime < serverMostRecentBreachDateTime) {
+    res.append("Last-Modified", serverMostRecentBreachDateTime);
+    res.json(HIBP.filterOutUnsafeBreaches(req.app.locals.breaches));
+  } else {
+    res.sendStatus(304);
+  }
+
+  if (new Date() - req.app.locals.breachesLoadedDateTime >= AppConstants.HIBP_RELOAD_BREACHES_TIMER) {
+    await HIBP.loadBreachesIntoApp(req.app);
```

Comment:
Ah, OK, I think I'm seeing how this API is used now...
But my QA-spidey senses are tingling, I'll need to troll through the code again and see how/when the images are being downloaded. I vaguely recall @lesleyjanenorton adding one of the missing logos in a recent PR, so I'm guessing we aren't automatically doing that on server startup or in the `loadBreachesIntoApp()` method. So basically this load breaches into the app, but just 404 the logos until we manually add new ones and push to stage/prod.

Or am I way off?